lms/djangoapps/experiments/tests/test_utils.py

```diff
@@ -116,3 +121,12 @@ def test_price_and_sku_from_multiple_courses(self):
         self.assertEqual(2, len(skus))
         self.assertIn(self.run_a_sku, skus)
         self.assertIn(self.entitlement_a_sku, skus)
+
+    def test_get_experiment_user_metadata_context(self):
+        course = CourseFactory.create(start=now() - timedelta(days=30))
+        user = UserFactory()
+        context = get_experiment_user_metadata_context(course, user)
+        user_metadata = context.get('user_metadata')
+
+        self.assertTrue(context['upgrade_price'], 'Free')
+        self.assertTrue(user_metadata['email'], 'robot+test+0@edx.org')
```

Comment:
The function we're testing just gets/sets fields into the context (and had no test before), so I kept it simple and check one of the fields set by the previous code and one of the fields set by the code I moved into there.